staging/src/k8s.io/kubectl/pkg/cmd/debug/debug.go

```diff
@@ -472,6 +472,18 @@ func (o *DebugOptions) visitNode(ctx context.Context, node *corev1.Node) (*corev
 //
 // visitPod returns a pod and debug container name for subsequent attach, if applicable.
 func (o *DebugOptions) visitPod(ctx context.Context, pod *corev1.Pod) (*corev1.Pod, string, error) {
+	// Display warning message if some capabilities are set by profile and non-root user is specified in .Spec.SecurityContext.RunAsUser.(#1650)
+	if o.Profile == ProfileGeneral || o.Profile == ProfileNetadmin || o.Profile == ProfileSysadmin ||
```

Comment:
I think, if we check the final pod template (after profiles and custom are applied), we can simplify these checks. For example, what about adding something like this in `debugByEphemeralContainer` function L:500;

```go
if o.Profile == ProfileNetadmin || o.Profile == ProfileSysadmin {
		if debugPod.Spec.SecurityContext != nil &&
			debugPod.Spec.SecurityContext.RunAsUser != nil &&
			*debugPod.Spec.SecurityContext.RunAsUser != 0 {
			fmt.Fprintf(o.ErrOut, "Warning: Non-root user is specified for the pod and some capabilities may not work. " +
				"Please consider using \"--custom\" with the custom profile specifying \"securityContext.runAsUser: 0\".")
		}
	}
```
and also similar functionality in `debugByCopy` L:624?